t/01-pdk/11-ctx.t

```diff
@@ -65,3 +65,69 @@ GET /t
 world
 --- no_error_log
 [error]
+
+
+
+=== TEST 3: ctx.plugin create namespace
+--- http_config eval: $t::Util::HttpConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local PDK = require "kong.pdk"
+            local pdk = PDK.new()
+            local plugin = { name = "plugin" }
+
+            pdk.ctx["plugin"] = plugin
+
+            pdk.ctx.plugin.hello = "world"
+            pdk.ctx.plugin.cats = {
+              "marry",
+              "suzie"
+            }
+
+            ngx.say(pdk.ctx.plugin.hello)
+            ngx.say(pdk.ctx.plugin.cats[1])
+            ngx.say(ngx.ctx.plugin)
+            ngx.say(ngx.ctx.hello)
+        }
+    }
+--- request
+GET /t
+--- response_body
+world
+marry
+nil
+nil
+--- no_error_log
+[error]
+
+
+
+=== TEST 4: ctx.plugin content is not overwritten between PDK instances
+--- http_config eval: $t::Util::HttpConfig
+--- config
+    location = /t {
+        content_by_lua_block {
+            local PDK = require "kong.pdk"
+            local pdk1 = PDK.new()
+            local pdk2 = PDK.new()
+            local plugin1 = { name = "plugin1" }
+            local plugin2 = { name = "plugin2" }
+
+            pdk1.ctx["plugin"] = plugin1
+            pdk1.ctx.plugin.hello = "hello"
+
+            pdk2.ctx["plugin"] = plugin2
+            pdk2.ctx.plugin.hello = "world"
+
+            pdk1.ctx["plugin"] = plugin1
+
+            ngx.say(pdk1.ctx.plugin.hello)
+        }
+    }
+--- request
+GET /t
+--- response_body
+hello
+--- no_error_log
+[error]
```

Comment:
This test does not seem particularly relevant to this issue either - I expect it to still pass even without your patch in this PR.